de.dashup.application/src/main/java/de/dashup/application/controllers/LayoutModeController.java

```diff
@@ -54,39 +47,15 @@ public String confirm(@RequestBody DashupPanelStructure body, Model model, HttpS
     public @ResponseBody
     ResponseEntity<Object> handleLayout(
             @CookieValue(name = "token", required = false) String token,
-            @RequestBody DashupPanelStructure dps,
+            @RequestBody Layout layout,
             HttpServletRequest request, HttpServletResponse response, Model model) throws SQLException {
-        User user = (User) this.localStorage.readObjectFromSession(request, "user");
-        if (user == null) return new ResponseEntity(HttpStatus.NETWORK_AUTHENTICATION_REQUIRED);
-
-        dps.getSections().sort(Comparator.comparingInt(DashupSectionStructure::getSectionOrder));
-        Iterator<DashupSectionStructure> iterator = dps.getSections().iterator();
-        DashupSectionStructure lastdss = null;
-        while (iterator.hasNext()) {
-            DashupSectionStructure dss = iterator.next();
-            if (dss.getSectionId().contains("sn")) {
-                if (lastdss == null) {
-                    DashupService.getInstance().addSection(user, dss.getSectionName(), -1, -1);
-                } else {
-                    DashupService.getInstance().addSection(user, dss.getSectionName(), Integer.valueOf(lastdss.getSectionId().substring(1)), -1);
-                }
-            } else {
-                if (dss.getSectionOrder() == -10) {
-                    DashupService.getInstance().deleteSection(user, Integer.valueOf(dss.getSectionId().substring(1)));
-                    iterator.remove();
-                } else {
-                    String section_name = dss.getSectionName();
-                    int section_id = Integer.valueOf(dss.getSectionId().substring(1));
-                    if (lastdss == null) {
-                        DashupService.getInstance().updateSection(user, section_name, section_id, -1, -1);
-                    } else {
-                        DashupService.getInstance().updateSection(user, section_name, section_id, Integer.valueOf(lastdss.getSectionId().substring(1)), -1);
-                    }
-                }
-            }
-            lastdss = dss;
+        DatabaseUser user = (DatabaseUser) this.localStorage.readObjectFromSession(request, "user");
```

Comment:
In order to display the widgets in the dashboard, I needed to generate basic logic fetching the data and generating a data structure for the central dashboard, which wasn't there yet. I changed all the dependencies of Sven's area, but as Sven decided the change the representation of the order of sections and widgets in the database, this would not be applicable anymore.